diff -up openssl-1.0.1e/crypto/cryptlib.c.disable-aesni openssl-1.0.1e/crypto/cryptlib.c
--- openssl-1.0.1e/crypto/cryptlib.c.disable-aesni	2013-02-11 16:26:04.000000000 +0100
+++ openssl-1.0.1e/crypto/cryptlib.c	2013-07-10 16:23:37.572078862 +0200
@@ -705,6 +705,10 @@ void OPENSSL_cpuid_setup(void)
     else
 	vec = OPENSSL_ia32_cpuid();
 
+    if (getenv("OPENSSL_DISABLE_AESNI") != NULL) {
+	vec = vec & ~(1ULL<<57); 
+    }
+
     /*
      * |(1<<10) sets a reserved bit to signal that variable
      * was initialized already... This is to avoid interference
