--- ./config.m4.old	2024-08-13 12:14:53.273367368 +0200
+++ ./config.m4	2024-08-13 12:20:46.644703516 +0200
@@ -2,7 +2,15 @@
 [  --enable-grpc           Enable grpc support])
 
 if test "$PHP_GRPC" != "no"; then
-  dnl Write more examples of tests here...
+  PHP_ARG_WITH(openssl-dir, for OpenSSL installation prefix,
+  [  --with-openssl-dir[=DIR]  Event: openssl installation prefix], yes, no)
+  PHP_SETUP_OPENSSL(GRPC_SHARED_LIBADD,
+  [
+  ], [
+    AC_MSG_ERROR([OpenSSL libraries not found.
+      Check whether openssl is on your PKG_CONFIG_PATH and the output in config.log)
+    ])
+  ])
 
   dnl # --with-grpc -> add include path
   PHP_ADD_INCLUDE(PHP_EXT_SRCDIR()/include)

