diff -Naupr twemproxy-0.4.1.orig/src/nc.c twemproxy-0.4.1/src/nc.c
--- twemproxy-0.4.1.orig/src/nc.c	2015-06-23 08:06:05.000000000 +0200
+++ twemproxy-0.4.1/src/nc.c	2016-06-23 17:12:51.431435925 +0200
@@ -28,7 +28,7 @@
 #include <nc_conf.h>
 #include <nc_signal.h>
 
-#define NC_CONF_PATH        "conf/twemproxy.yml"
+#define NC_CONF_PATH        "/etc/twemproxy.yml"
 
 #define NC_LOG_DEFAULT      LOG_NOTICE
 #define NC_LOG_MIN          LOG_EMERG
