--- bugzilla-4.0.1/Bugzilla/Constants.pm	2011-08-05 12:14:57.301671513 +0200
+++ bugzilla-4.0.1-rw/Bugzilla/Constants.pm	2011-08-05 12:16:51.798741609 +0200
@@ -605,18 +605,18 @@ sub bz_locations {
         'cgi_path'    => $libpath,
         'templatedir' => "$libpath/template",
         'project'     => $project,
-        'localconfig' => "$libpath/$localconfig",
-        'datadir'     => "$libpath/$datadir",
-        'attachdir'   => "$libpath/$datadir/attachments",
+        'localconfig' => "/etc/bugzilla/localconfig",
+        'datadir'     => "/var/lib/bugzilla/$datadir",
+        'attachdir'   => "/var/lib/bugzilla/$datadir/attachments",
         'skinsdir'    => "$libpath/skins",
-        'graphsdir'   => "$libpath/graphs",
+        'graphsdir'   => "/var/lib/bugzilla/graphs",
         # $webdotdir must be in the web server's tree somewhere. Even if you use a 
         # local dot, we output images to there. Also, if $webdotdir is 
         # not relative to the bugzilla root directory, you'll need to 
         # change showdependencygraph.cgi to set image_url to the correct 
         # location.
         # The script should really generate these graphs directly...
-        'webdotdir'   => "$libpath/$datadir/webdot",
+        'webdotdir'   => "/var/lib/bugzilla/$datadir/webdot",
         'extensionsdir' => "$libpath/extensions",
     };
 }
