diff -up openssl-fips-0.9.8e/ssl/ssl_algs.c.sha256 openssl-fips-0.9.8e/ssl/ssl_algs.c
--- openssl-fips-0.9.8e/ssl/ssl_algs.c.sha256	2010-12-07 17:45:32.000000000 +0100
+++ openssl-fips-0.9.8e/ssl/ssl_algs.c	2011-03-07 18:58:56.000000000 +0100
@@ -104,6 +104,14 @@ int SSL_library_init(void)
 	EVP_add_digest_alias(SN_sha1,"ssl3-sha1");
 	EVP_add_digest_alias(SN_sha1WithRSAEncryption,SN_sha1WithRSA);
 #endif
+#ifndef OPENSSL_NO_SHA256
+	EVP_add_digest(EVP_sha224());
+	EVP_add_digest(EVP_sha256());
+#endif
+#ifndef OPENSSL_NO_SHA512
+	EVP_add_digest(EVP_sha384());
+	EVP_add_digest(EVP_sha512());
+#endif
 #if !defined(OPENSSL_NO_SHA) && !defined(OPENSSL_NO_DSA)
 	EVP_add_digest(EVP_dss1()); /* DSA with sha1 */
 	EVP_add_digest_alias(SN_dsaWithSHA1,SN_dsaWithSHA1_2);
