diff -Naupr mariadb-10.0.34.orig/scripts/mysql_install_db.pl.in mariadb-10.0.34/scripts/mysql_install_db.pl.in
--- mariadb-10.0.34.orig/scripts/mysql_install_db.pl.in	2018-01-27 19:40:37.000000000 +0100
+++ mariadb-10.0.34/scripts/mysql_install_db.pl.in	2018-02-12 16:34:53.186118648 +0100
@@ -318,7 +318,7 @@ elsif ( $opt->{basedir} )
                     find_in_basedir($opt,"file","mysqld-nt",
                                   "bin");  # ,"sql"
   $pkgdatadir     = find_in_basedir($opt,"dir","fill_help_tables.sql",
-                                    "share","share/mysql");  # ,"scripts"
+                                    "share","@INSTALL_MYSQLSHAREDIR@");  # ,"scripts"
   $scriptdir      = "$opt->{basedir}/scripts";
 }
 else
