diff -Naupr ClickHouse-1.1.54030-stable.orig/contrib/libpoco/Net/CMakeLists.txt ClickHouse-1.1.54030-stable/contrib/libpoco/Net/CMakeLists.txt
--- ClickHouse-1.1.54030-stable.orig/contrib/libpoco/Net/CMakeLists.txt	2016-10-24 14:34:08.000000000 +0200
+++ ClickHouse-1.1.54030-stable/contrib/libpoco/Net/CMakeLists.txt	2016-11-07 11:55:22.420160958 +0100
@@ -17,7 +17,7 @@ if(WIN32)
         set(SYSLIBS ${SYSLIBS} "ws2_32.lib" "iphlpapi.lib")
     endif()    
 else()
-    set(SYSLIBS ${SYSLIBS} "libanl.a")
+    set(SYSLIBS ${SYSLIBS} "libanl.so")
 endif(WIN32)
 
 add_library( "${LIBNAME}" ${LIB_MODE} ${SRCS} )
diff -Naupr ClickHouse-1.1.54030-stable.orig/dbms/CMakeLists.txt ClickHouse-1.1.54030-stable/dbms/CMakeLists.txt
--- ClickHouse-1.1.54030-stable.orig/dbms/CMakeLists.txt	2016-10-24 14:34:08.000000000 +0200
+++ ClickHouse-1.1.54030-stable/dbms/CMakeLists.txt	2016-11-07 11:55:22.421160948 +0100
@@ -12,7 +12,7 @@ endif()
 if (DISABLE_MONGODB)
 	add_definitions(-D DISABLE_MONGODB)
 else()
-	set (LINK_MONGOCLIENT libmongoclient.a libssl.a libcrypto.a libboost_thread.a)
+	set (LINK_MONGOCLIENT libmongoclient.a ssl crypto libboost_thread.a)
 endif()
 
 if (DISABLE_LIBTCMALLOC)
@@ -977,7 +977,7 @@ target_link_libraries(dbms
 	double-conversion
 	${LINK_LIBRARIES_ONLY_ON_X86_64}
 	re2 re2_st
-	libcrypto.a
+	crypto
 	libboost_system.a
 	${LINK_MONGOCLIENT}
 	libboost_regex.a
diff -Naupr ClickHouse-1.1.54030-stable.orig/libs/libcommon/CMakeLists.txt ClickHouse-1.1.54030-stable/libs/libcommon/CMakeLists.txt
--- ClickHouse-1.1.54030-stable.orig/libs/libcommon/CMakeLists.txt	2016-10-24 14:34:08.000000000 +0200
+++ ClickHouse-1.1.54030-stable/libs/libcommon/CMakeLists.txt	2016-11-07 11:55:22.421160948 +0100
@@ -64,9 +64,9 @@ target_link_libraries (
 	pocoext
 	${MALLOC_LIBRARIES}
 	pthread
-	libglib-2.0.a
-	libicui18n.a libicuuc.a libicudata.a
-	librt.a)
+	glib-2.0
+	icui18n icuuc icudata
+	rt)
 
 IF(TESTS)
 	add_subdirectory (src/tests)
diff -Naupr ClickHouse-1.1.54030-stable.orig/libs/libcommon/src/tests/CMakeLists.txt ClickHouse-1.1.54030-stable/libs/libcommon/src/tests/CMakeLists.txt
--- ClickHouse-1.1.54030-stable.orig/libs/libcommon/src/tests/CMakeLists.txt	2016-10-24 14:34:08.000000000 +0200
+++ ClickHouse-1.1.54030-stable/libs/libcommon/src/tests/CMakeLists.txt	2016-11-07 11:55:22.421160948 +0100
@@ -8,10 +8,10 @@ add_executable (multi_version multi_vers
 add_executable (json_test json_test.cpp)
 add_executable (strong_typedef strong_typedef.cpp)
 
-target_link_libraries (date_lut_init common libicui18n.a libicuuc.a libicudata.a dl)
-target_link_libraries (date_lut2 common libicui18n.a libicuuc.a libicudata.a dl)
-target_link_libraries (date_lut3 common libicui18n.a libicuuc.a libicudata.a dl)
-target_link_libraries (date_lut4 common libicui18n.a libicuuc.a libicudata.a dl)
+target_link_libraries (date_lut_init common icui18n icuuc icudata dl)
+target_link_libraries (date_lut2 common icui18n icuuc icudata dl)
+target_link_libraries (date_lut3 common icui18n icuuc icudata dl)
+target_link_libraries (date_lut4 common icui18n icuuc icudata dl)
 target_link_libraries (multi_version dbms libboost_system.a rt)
 target_link_libraries (json_test dbms libboost_system.a rt)
 target_link_libraries (strong_typedef common)
diff -Naupr ClickHouse-1.1.54030-stable.orig/libs/libmysqlxx/CMakeLists.txt ClickHouse-1.1.54030-stable/libs/libmysqlxx/CMakeLists.txt
--- ClickHouse-1.1.54030-stable.orig/libs/libmysqlxx/CMakeLists.txt	2016-10-24 14:34:08.000000000 +0200
+++ ClickHouse-1.1.54030-stable/libs/libmysqlxx/CMakeLists.txt	2016-11-07 11:55:22.421160948 +0100
@@ -26,15 +26,15 @@ add_library (mysqlxx
 
 add_dependencies (mysqlxx common)
 
-target_link_libraries (mysqlxx common libmysqlclient.a libssl.a libcrypto.a libz.a dl)
+target_link_libraries (mysqlxx common libmysqlclient.a ssl crypto z dl)
 
 
-ADD_CUSTOM_COMMAND(
-    TARGET mysqlxx
-    POST_BUILD
-    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/patch.sh
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-    COMMENT "Patching mysqlclient library.")
+#ADD_CUSTOM_COMMAND(
+#    TARGET mysqlxx
+#    POST_BUILD
+#    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/patch.sh
+#    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+#    COMMENT "Patching mysqlclient library.")
 
 IF(TESTS)
 	add_subdirectory (src/tests)
