diff -Naupr ClickHouse-1.1.54022-stable.orig/CMakeLists.txt ClickHouse-1.1.54022-stable/CMakeLists.txt
--- ClickHouse-1.1.54022-stable.orig/CMakeLists.txt	2016-09-26 13:40:02.000000000 +0000
+++ ClickHouse-1.1.54022-stable/CMakeLists.txt	2016-10-06 14:11:02.451016158 +0000
@@ -144,6 +144,7 @@ ENDIF()
 include_directories (/usr/local/include/)
 
 link_directories (/usr/local/lib)
+link_directories (/usr/lib64/mysql)
 
 # Directory for Yandex specific files
 SET(CLICKHOUSE_PRIVATE_DIR ${ClickHouse_SOURCE_DIR}/private/)
